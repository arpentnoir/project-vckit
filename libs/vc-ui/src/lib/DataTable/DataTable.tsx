import AddIcon from '@mui/icons-material/Add';
import { Box, BoxProps, Stack } from '@mui/material';
import {
  DataGrid,
  DataGridProps,
  GridToolbarQuickFilter,
} from '@mui/x-data-grid';
import { Button } from '..';

const SimpleFilter = ({
  label,
  handleAction,
}: {
  label: string;
  handleAction: () => void;
}) => {
  return (
    <Stack
      padding="16px"
      direction={{ xs: 'column', sm: 'row' }}
      spacing="16px"
      justifyContent="space-between"
    >
      <GridToolbarQuickFilter />

      {label && handleAction && (
        <Button
          color="error"
          label={label}
          leftIcon={<AddIcon />}
          onClick={handleAction}
          style={{ marginLeft: 'auto' }}
        />
      )}
    </Stack>
  );
};

interface IDataTable extends DataGridProps {
  handleAction?: () => void;
  toolBarActionLabel?: string;
  containerProps?: BoxProps;
}

export const DataTable = ({
  handleAction,
  toolBarActionLabel,
  containerProps,
  ...rest
}: IDataTable) => {
  return (
    <Box style={{ height: 400, width: '100%' }} {...containerProps}>
      <DataGrid
        components={{
          Toolbar: SimpleFilter,
        }}
        componentsProps={{
          toolbar: { handleAction, label: toolBarActionLabel },
        }}
        {...rest}
      />
    </Box>
  );
};
