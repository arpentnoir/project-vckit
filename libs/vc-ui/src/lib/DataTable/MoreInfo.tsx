import MoreVertIcon from '@mui/icons-material/MoreVert';
import { ButtonBase, Menu, MenuItem, Stack } from '@mui/material';
import { GridCellParams } from '@mui/x-data-grid';
import React from 'react';

interface IMoreInfo {
  items: { [key: string]: () => void };
  params: GridCellParams;
}

export const MoreInfo = ({ items, params }: IMoreInfo) => {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.stopPropagation();
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <Stack
      direction="row"
      style={{ justifyContent: 'end', marginLeft: 'auto' }}
    >
      <ButtonBase
        onClick={handleClick}
        data-testid={`more-info-menu-button:${params.id}`}
      >
        <MoreVertIcon />
      </ButtonBase>
      <Menu
        id={`basic-menu:${params.id}`}
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
      >
        {Object.entries(items).map(([name, handleAction]) => {
          return (
            <MenuItem
              data-testid={`more-info-list-item:${name}`}
              key={name}
              onClick={handleAction}
            >
              {name}
            </MenuItem>
          );
        })}
      </Menu>
    </Stack>
  );
};
