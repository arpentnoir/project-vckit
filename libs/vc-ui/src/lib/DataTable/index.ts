export * from './DataTable';
export * from './MoreInfo';
