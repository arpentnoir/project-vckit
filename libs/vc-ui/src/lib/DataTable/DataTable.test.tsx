import { GridColDef } from '@mui/x-data-grid';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { DataTable } from './DataTable';

const columns: GridColDef[] = [
  {
    field: 'id',
    headerName: 'ID',
  },
  {
    field: 'firstName',
    headerName: 'First Name',
  },
  {
    field: 'lastName',
    headerName: 'Last Name',
  },
];

const rows = [
  {
    id: '006',
    firstName: 'Postman',
    lastName: 'Pat',
  },
  {
    id: '007',
    firstName: 'Wreck-It',
    lastName: 'Ralph',
  },
];

const mockHandleFunction = jest.fn();

describe('DataTable', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <DataTable
        handleAction={mockHandleFunction}
        columns={columns}
        rows={rows}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should display all fields and their values', () => {
    const { getByText } = render(
      <DataTable
        handleAction={mockHandleFunction}
        columns={columns}
        rows={rows}
      />
    );

    getByText('006');
    getByText('Postman');
    getByText('Pat');

    getByText('007');
    getByText('Wreck-It');
    getByText('Ralph');
  });

  it('should display relevant results from search', async () => {
    const { getByPlaceholderText, queryByText } = render(
      <DataTable
        handleAction={mockHandleFunction}
        columns={columns}
        rows={rows}
      />
    );

    expect(queryByText('007')).toBeTruthy();
    expect(queryByText('Wreck-It')).toBeTruthy();
    expect(queryByText('Ralph')).toBeTruthy();

    const searchBar = getByPlaceholderText('Searchâ€¦');

    fireEvent.change(searchBar, { target: { value: '006' } });

    await waitFor(() => {
      expect(queryByText('007')).toBeFalsy();
      expect(queryByText('Wreck-It')).toBeFalsy();
      expect(queryByText('Ralph')).toBeFalsy();
    });
  });
});
