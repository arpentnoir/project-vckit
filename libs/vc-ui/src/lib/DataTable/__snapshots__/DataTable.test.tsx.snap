// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
      style="height: 400px; width: 100%;"
    >
      <div
        aria-colcount="3"
        aria-multiselectable="false"
        aria-rowcount="3"
        class="MuiDataGrid-root MuiDataGrid-root--densityStandard css-1e2bxag-MuiDataGrid-root"
        role="grid"
      >
        <div>
          <div>
            <div
              class="css-qz6wf1-MuiStack-root"
            >
              <div
                aria-label="Search"
                class="MuiFormControl-root MuiTextField-root css-3be3ve-MuiFormControl-root-MuiTextField-root-MuiDataGrid-toolbarQuickFilter"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1ptx2yq-MuiInputBase-root-MuiInput-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-c63i49-MuiInputBase-input-MuiInput-input"
                    id=":r0:"
                    placeholder="Searchâ€¦"
                    type="search"
                    value=""
                  />
                  <button
                    aria-label="Clear"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-fp9tcu-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="CloseIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-gl260s-MuiDataGrid-columnHeadersInner"
              role="rowgroup"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="1"
                class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                role="row"
              >
                <div
                  aria-colindex="1"
                  aria-label="ID"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                  data-field="id"
                  role="columnheader"
                  style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                  tabindex="0"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          ID
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-controls=":r3:"
                        aria-haspopup="true"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        id=":r4:"
                        tabindex="-1"
                        title="Menu"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-label="First Name"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                  data-field="firstName"
                  role="columnheader"
                  style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          First Name
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-controls=":r6:"
                        aria-haspopup="true"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        id=":r7:"
                        tabindex="-1"
                        title="Menu"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="3"
                  aria-label="Last Name"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                  data-field="lastName"
                  role="columnheader"
                  style="height: 56px; width: 100px; min-width: 100px; max-width: 100px;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          Last Name
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiDataGrid-menuIcon"
                    >
                      <button
                        aria-controls=":r9:"
                        aria-haspopup="true"
                        aria-label="Menu"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        id=":ra:"
                        tabindex="-1"
                        title="Menu"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                          data-testid="TripleDotsVerticalIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 56px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="overflow: visible; height: 0px; width: 0px;"
          >
            <div
              class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
              style="width: 0px; height: auto; margin-top: 56px; overflow-x: hidden;"
            >
              <div
                class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                style="width: auto; height: 104px; min-height: auto;"
              >
                <div
                  class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                  style="transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-rowindex="2"
                    aria-selected="false"
                    class="MuiDataGrid-row"
                    data-id="006"
                    data-rowindex="0"
                    role="row"
                    style="max-height: 52px; min-height: 52px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="id"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        006
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="firstName"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        Postman
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="lastName"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        Pat
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="3"
                    aria-selected="false"
                    class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                    data-id="007"
                    data-rowindex="1"
                    role="row"
                    style="max-height: 52px; min-height: 52px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="id"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        007
                      </div>
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="firstName"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        Wreck-It
                      </div>
                    </div>
                    <div
                      aria-colindex="3"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="2"
                      data-field="lastName"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      >
                        Ralph
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
        <div>
          <div
            class="MuiDataGrid-footerContainer css-17jjc08-MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                  id=":r2:"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby=":r2: :r1:"
                    class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                    id=":r1:"
                    role="button"
                    tabindex="0"
                  >
                    100
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="100"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  1â€“2 of 2
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to previous page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowLeftIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to next page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
